### PR TITLE
Doc/Core#666 - Move GlobalAppConfig attributes to CoreSection

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-core/issues/666

### CODE REVIEW

**Comment by jrobinAV** [comment]
@erikfTW I propose to add you as a reviewer of the documentation pull requests so that you are notified and can follow the topics in progress and the parts that are evolving. Of course, it s not mandatory to review them if you don't need to. Feel free to ignore them.

File: docs/manuals/core/config/config-checker.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ exception if at least one issue collected has the `ERROR` level.
 
 Here is the list of the possible issues that could be returned by the checker:
 
-- An `ERROR` issue is created if the `clean_entities_enabled` property is populated in the `GlobalAppConfig^` with a
-  non-Boolean value.
+- An `WARNING` issue is created if the `repository_type` property is populated in the `CoreSection^` with an
```
Comment (jrobinAV) [inline docs/manuals/core/config/config-checker.md]:
```suggestion
- A `WARNING` issue is created if the `repository_type` property is populated in the `CoreSection^` with an
```

File: docs/manuals/core/config/config.md [hunk 2]
```diff
@@ -5,39 +5,45 @@ using the following import:
...
 
     Config.scenarios
     ```
-- the [job configuration](job-config.md) as a `JobConfig^`:
+- The [Job configuration](job-config.md) as a `JobConfig^`:
     ```python linenums="1"
     from taipy import Config
 
     Confi.job_config
     ```
```
Comment (jrobinAV) [inline docs/manuals/core/config/config.md:39]: Nothing about the migration section? I guess it will be part of another PR. Right?

File: docs/manuals/core/config/core-config.md [hunk 3]
```diff
@@ -1,18 +1,12 @@
-The `GlobalAppConfig^` holds configuration fields related to the global application.
+The `CoreSection^` holds configuration fields related to the Core application.
```
Comment (jrobinAV) [inline docs/manuals/core/config/core-config.md]:
```suggestion
The `CoreSection^` holds configuration fields related to the Core service.
```

File: docs/manuals/core/config/core-config.md [hunk 4]
```diff
@@ -1,18 +1,12 @@
...
+- _**root_folder**_: The path of the base folder for the Taipy application, its default value is "./taipy/".
 - _**storage_folder**_: The folder name used to store Taipy data, its default value is ".data/". It is used in
   conjunction with the root_folder field. That means the default storage path is "./taipy/.data/".
-- _**clean_entities_enabled**_: The field to activate/deactivate the clean entities feature.
-  Its default value is `ENV[TAIPY_CLEAN_ENTITIES_ENABLED]:bool` meaning that the default value is read from the
-  `TAIPY_CLEAN_ENTITIES_ENABLED` environment variable. If the environment variable is not set, the default value is
-  False. <br>
-  Since it is risky to delete all entities on a production environment, Taipy proposes a way to activate this
-  feature only on specific environments. That is why the default value points to an environment variable.
 - _**read_entity_retry**_: The integer number of retry when reading an entity after a failed attempt (in case of
```
Comment (jrobinAV) [inline docs/manuals/core/config/core-config.md]:
```suggestion
- _**read_entity_retry**_: An integer number only used with `filesystem` _**repository_type**_. <br>
It corresponds to the number of times Taipy retries reading an entity after a failed attempt for concurrent access. <br>
The default value is 3.
```

File: docs/manuals/core/config/core-config.md [hunk 5]
```diff
@@ -1,18 +1,12 @@
...
-  concurrent access). The default value is 0.
+  concurrent access). The default value is 3.
```
Comment (jrobinAV) [inline docs/manuals/core/config/core-config.md]:
```suggestion
```

File: docs/manuals/core/config/core-config.md [hunk 6]
```diff
@@ -21,22 +15,31 @@ Here are the (optional) configurable properties:
...
   _**mongodb_password**_, _**mongodb_port**_, _**application_db**_, _**properties**_. Please refer to
-  [MongoDB storage section](global-config.md#mongodb-storage-for-taipy-entities).
+  [MongoDB storage section](core-config.md#mongodb-storage-for-taipy-entities).
+- _**mode**_: A string that indicates the mode of the version management system.
+  Possible values are *"development"*, *"experiment"*, or *"production"*.
+- _**version_number**_: The identifier of the version. In development mode, the version number is ignored.
+- _**force**_: Indicates whether Taipy will override a version even if the configuration has changed
+    or not and run the application. Default to False.
+- _**clean_entities**_: Indicates whether running a Taipy Core service will clean all current version
+    entities or not before running the application. Default to False.
```
Comment (jrobinAV) [inline docs/manuals/core/config/core-config.md:29]: Sorry, I missed that in the other PRs for the ticket [core-#666](https://github.com/Avaiga/taipy-core/issues/666). Probably because it s a Demoniac ticket. 🤣

In the ticket, the field `clean_entities` should be completely removed from the config.
Since we have the development mode that does it automatically, and since we have the CLI to manage versions, it is not needed anymore. Plus, it is error-prone and can be dangerous to remove all production entities just with a config value.
I discussed with @FlorianJacta that agrees with that.

That means we need to remove the attribute from the `CoreSection`, and it also means we need to remove lines 205, 206, and 207 in `taipy-core/src/taipy/core/taipy.py`

```suggestion
```

**Comment by erikfTW** [comment]
> @erikfTW I propose to add you as a reviewer of the documentation pull requests so that you are notified and can follow the topics in progress and the parts that are evolving. Of course, it s not mandatory to review them if you don't need to. Feel free to ignore them.

Yes, this is a good idea, at least I can keep track of the documentation changes, thanks!

File: docs/relnotes.md [hunk 7]
```diff
@@ -30,6 +30,9 @@ Not published yet.
 3.0.0
 
 - `Scope.PIPELINE` has been removed from `Scope^` values.
+- The `root_folder`, `storage_folder`, `read_entity_retry`, `repository_type`, and `repository_properties`
```
Comment (jrobinAV) [inline docs/relnotes.md:33]:
```suggestion
- The `root_folder`, `storage_folder`, `repository_type`, and `repository_properties`
```

**Review by jrobinAV — CHANGES_REQUESTED** [review]

File: docs/manuals/core/config/config-checker.md [hunk 8]
```diff
@@ -13,8 +13,8 @@ exception if at least one issue collected has the `ERROR` level.
...
+  unsupported repository value.
 - An `ERROR` issue is created if the `storage_type` and the `scope` properties of any `DataNodeConfig^` have not
```
Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md:18]: An `ERROR` issue is created if the `storage_type` and the `scope` properties of any `DataNodeConfig^` are not

File: docs/manuals/core/config/config-checker.md [hunk 9]
```diff
@@ -13,8 +13,8 @@ exception if at least one issue collected has the `ERROR` level.
...
   been provided with a correct value.
```
Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md:19]: provided with a correct value.

Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md:19]: provided with a correct value.

File: docs/manuals/core/config/config-checker.md (hunk 8 above)
Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md]:
```suggestion
- An `ERROR` issue is created if the `storage_type` and the `scope` properties of any `DataNodeConfig^` are not
```

File: docs/manuals/core/config/config-checker.md (hunk 9 above)
Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md]:
```suggestion
  provided with a correct value.
```

File: docs/manuals/core/config/config-checker.md [hunk 10]
```diff
@@ -13,8 +13,8 @@ exception if at least one issue collected has the `ERROR` level.
 
 Here is the list of the possible issues that could be returned by the checker:
```
Comment (erikfTW) [inline docs/manuals/core/config/config-checker.md]:
```suggestion
Here is the list of the possible issues that the checker could return:
```

File: docs/manuals/core/config/config.md (hunk 2 above)
Comment (trgiangdo) [reply docs/manuals/core/config/config.md:39]: Yes, I think it should be a part of the Data Migration API doc.

File: docs/relnotes.md (hunk 7 above)
Comment (trgiangdo) [reply docs/relnotes.md:33]: The `read_entity_retry` was on `GlobalAppConfig` before. It was also documented, so I think we should specify it here.

File: docs/manuals/core/config/core-config.md (hunk 6 above)
Comment (trgiangdo) [reply docs/manuals/core/config/core-config.md:29]: We need to update some example, and the code on taipy-core for this also. Do you want to do it in another ticket?

Comment (jrobinAV) [reply docs/manuals/core/config/core-config.md:29]: I don't mind. It s up to you.

File: docs/relnotes.md (hunk 7 above)
Comment (jrobinAV) [reply docs/relnotes.md:33]: Absolutely.

**Review by jrobinAV — APPROVED** [review]

File: docs/manuals/core/config/core-config.md (hunk 6 above)
Comment (trgiangdo) [reply docs/manuals/core/config/core-config.md:29]: Ticket created: https://github.com/Avaiga/taipy-core/issues/693
